include/triggeralgs/TriggerActivityFactory.hpp

```diff
@@ -0,0 +1,43 @@
+/* @file: TriggerActivityFactory.hpp
+ *
+ * This is part of the DUNE DAQ Application Framework, copyright 2023.
+ * Licensing/copyright details are in the COPYING file that you should have
+ * received with this code.
+ */
+
+#ifndef TRIGGERALGS_TRIGGER_ACTIVITY_FACTORY_HPP_
+#define TRIGGERALGS_TRIGGER_ACTIVITY_FACTORY_HPP_
+
+#include <memory>
+#include <string>
+#include <unordered_map>
+#include "triggeralgs/TriggerActivityMaker.hpp"
+
+#define REGISTER_TRIGGER_ACTIVITY_MAKER(tam_name, tam_class) \
+  bool tam_class::s_registered = TriggerActivityFactory::registerCreator(tam_name, []()->std::shared_ptr<TriggerActivityMaker> {return std::make_shared<tam_class>();});
+
+namespace triggeralgs {
+
+class TriggerActivityFactory
+{
+  public:
+    using TAMakerCreator = std::shared_ptr<TriggerActivityMaker>(*)();
```

Comment:
`using TAMakerCreator = std::function<std::unique_ptr<TriggerActivityMaker>()>>` would probably make more sense here, rather than the double bracket with a star.

These should be unique pointers too, since TAMs are not shared between different objects.